### PR TITLE
[DOCS] added sending events section in docs

### PR DESCRIPTION
**Pull request by zakisk**
**Notes for Reviewers**

This PR adds documentation for "How to send events from Golang".

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 0ecee0144b3c9cca7057a3e2efe2e0af24ab89da at: https://652438c3da108e092f6773a6--meshery-docs-preview.netlify.app

**Comment by saurabh100ni** [comment]
Thanks for raising the PR, let's Discuss this in the Meshery Development Call today,
Please add this as an agenda in the Doc  : [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)

File: docs/_data/toc.yml [hunk 1]
```diff
@@ -159,6 +159,8 @@
       url: guides/troubleshooting
     - title: Upgrading Meshery
       url: guides/upgrade
+    - title: Send Events From Meshery Server Using Golang
```
Comment (leecalcote) [inline docs/_data/toc.yml:162]: Move this to the Contributing section, please.

Rename to follow the Contributing Doc page naming convention.

File: docs/pages/guides/send-event-in-server.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+layout: default
+title: Send Events From Meshery Server Using Golang
+description: This guide is to help backend contributors how to send server events using Golang.
+permalink: guides/send-events
+type: Guides
+language: en
+---
+
+&emsp;&emsp;&emsp;Meshery incorporates an internal events publication mechanism that serves to provide users with real-time updates on the processes occuring within the Meshery API when they interact with its endpoints. it ensures that users are kept in the loop regarding the ongoing activities within the API, and guides users towards future steps to resolve issues. This guide will provide step-by-step instructions on how to send events from the server codebase, including when to trigger events and what information to include in.
```
Comment (leecalcote) [inline docs/pages/guides/send-event-in-server.md:10]:
```suggestion
Meshery incorporates an internal events publication mechanism that serves to provide users with real-time updates on the processes occuring within the Meshery API when they interact with its endpoints. it ensures that users are kept in the loop regarding the ongoing activities within the API, and guides users towards future steps to resolve issues. This guide will provide step-by-step instructions on how to send events from the server codebase, including when to trigger events and what information to include in.
```

File: docs/pages/guides/send-event-in-server.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                  .WithAction(string)
+                  .WithSeverity(EventSeverity)
+                  .WithDescription(string)
+                  .WithMetadata(map[string]interface{})
+                  .Build()
+  </div>
+ </pre>
+&nbsp;&nbsp;
+
+*Note: events is a package **(github.com/layer5io/meshkit/models/events)** from meshkit containing source code of event related functionality*
```
Comment (leecalcote) [inline docs/pages/guides/send-event-in-server.md:31]:
```suggestion
*Note: `events` is a package [github.com/meshery/meshkit/models/events](https://github.com/meshery/meshkit) from MeshKit containing source code of event related functionality*
```

File: docs/pages/guides/send-event-in-server.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+If `provider.GetK8sContext` succeeds, an event with a [Success](https://github.com/meshery/meshkit/blob/ea3c60907a1cd1902902a4113206579992772083/models/events/events.go#L21) severity level is constructed. The event includes a success message and metadata containing the context name and the address of the Kubernetes API server, obtained from the operation. This information aims to provide users with relevant details about the successful operation.
+
+after constructing the Event object using the Build function, event is stored into database for records and subsequently published in a new goroutine through [EventBroadcaster](https://github.com/meshery/meshery/blob/995ab671a12013088f874430cfa2c0f025b073d2/server/models/handlers.go#L241) member variable of [HandlerConfig](https://github.com/meshery/meshery/blob/995ab671a12013088f874430cfa2c0f025b073d2/server/models/handlers.go#L206), which is part of [Handler](https://github.com/meshery/meshery/blob/995ab671a12013088f874430cfa2c0f025b073d2/server/handlers/handler_instance.go#L18) instance.
+
```
Comment (leecalcote) [inline docs/pages/guides/send-event-in-server.md:107]: @ptdatta have you documented the Notification Formatters, yet? A link and description of them needs to be included here.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: docs/pages/guides/send-event-in-server.md (hunk 4 above)
Comment (ptdatta) [reply docs/pages/guides/send-event-in-server.md:107]: No Currently I am working on the meshmodel UI. OK I'll do it by today.

Comment (ptdatta) [reply docs/pages/guides/send-event-in-server.md:107]: @leecalcote  Done✅ [#9086](https://github.com/meshery/meshery/pull/9086)

File: docs/pages/guides/send-event-in-server.md (hunk 2 above)
Comment (MUzairS15) [inline docs/pages/guides/send-event-in-server.md:10]:
``` suggestion
&emsp;&emsp;&emsp;Meshery incorporates an internal events publication mechanism that provides users with real-time updates on the processes occurring within the Meshery server when interacting with its endpoints. It ensures that users are kept in the loop regarding the ongoing activities within the API, and guides users towards future steps to resolve issues. This guide will provide step-by-step instructions on sending events from the server, including when to trigger events and what information to include.
```

File: docs/pages/guides/send-event-in-server.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	eventBuilder := events.NewEvent().FromUser(userID).FromSystem(*h.SystemID).WithCategory("connection").WithAction("ping")
+	if connectionID != "" {
+		// Get the context associated with this ID
+		k8sContext, err := provider.GetK8sContext(token, connectionID)
+		eventBuilder.ActedUpon(uuid.FromStringOrNil(connectionID))
+		if err != nil {
+			eventBuilder.WithSeverity(events.Error).WithDescription(fmt.Sprintf("Encountered an issue while retrieving kubernetes context for connection ID `%s`", connectionID)).
+				WithMetadata(map[string]interface{}{
+					"error":                 err,
+					"probable_cause":        "Meshery might not be able to access Kubeconfig.",
```
Comment (MUzairS15) [inline docs/pages/guides/send-event-in-server.md:75]: This is incorrect @zakisk, the severity of the event is an error, therefore the error should be MeshKit error, and the error would contain `probable_cause` and `suggested_remediation`.

Please update the example. Or add a snippet/link from the existing codebase.
When you add a link make sure to not source from the master branch but instead from one of the release tags. As in future, there will be changes in the code, which would result in invalid reference.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: docs/pages/guides/send-event-in-server.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
+---
+layout: default
+title: Send Events From Meshery Server Using Golang
+description: This guide is to help backend contributors how to send server events using Golang.
+permalink: guides/send-events
+type: Guides
+language: en
+---
+
+&emsp;&emsp;&emsp;Meshery incorporates an internal events publication mechanism that provides users with real-time updates on the processes occurring within the Meshery server when interacting with its endpoints. It ensures that users are kept in the loop regarding the ongoing activities within the API, and guides users towards future steps to resolve issues. This guide will provide step-by-step instructions on sending events from the server, including when to trigger events and what information to include.
```
Comment (leecalcote) [inline docs/pages/guides/send-event-in-server.md:10]:
```suggestion
Meshery incorporates an internal events publication mechanism that provides users with real-time updates on the processes occurring within the Meshery server when interacting with its endpoints. It ensures that users are kept in the loop regarding the ongoing activities within the API, and guides users towards future steps to resolve issues. This guide will provide step-by-step instructions on sending events from the server, including when to trigger events and what information to include.
```

**Comment by l5io** [comment]
🚀 Preview for commit e544f4cd83ffe1e3a05efbbde92ff1e4ece6aa53 at: https://65293f834b271a3725634ed1--meshery-docs-preview.netlify.app

File: docs/pages/guides/send-event-in-server.md (hunk 4 above)
Comment (hamza-mohd) [reply docs/pages/guides/send-event-in-server.md:107]: Thanks, @ptdatta

**Comment by l5io** [comment]
🚀 Preview for commit 2ea5b8733cc8230b24a21196d1673d361bda9b89 at: https://652b8bfbd80c343ef5c1db6a--meshery-docs-preview.netlify.app

File: docs/pages/project/contributing/contributing-sever-events.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
+---
+layout: default
+title: Contributing to Server Events Using Golang
```
Comment (leecalcote) [inline docs/pages/project/contributing/contributing-sever-events.md:3]:
```suggestion
title: Contributing to Meshery Server Events
```

**Review by leecalcote — APPROVED** [review]

File: docs/_data/toc.yml [hunk 8]
```diff
@@ -29,6 +29,8 @@
           url: project/contributing/contributing-cypress
         - title: Meshery Docker Extension
           url: project/contributing/contributing-docker-extension
+        - title: Server Events (Golang)
+          url: project/contributing/send-events
```
Comment (leecalcote) [inline docs/_data/toc.yml:33]:
```suggestion
        - title: Meshery Server Events
          url: project/contributing-server-events
```

File: docs/pages/project/contributing/contributing-sever-events.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+description: This guide is to help backend contributors how to send server events using Golang.
+permalink: project/contributing/send-events
```
Comment (leecalcote) [inline docs/pages/project/contributing/contributing-sever-events.md:5]:
```suggestion
permalink: project/contributing/contributing-server-events
```

**Pull request merged**